docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,546 @@
+---
+layout: docs
+---
+
+# Introduction
+
+Oftentimes, our applications read and manipulate data from a variety of
+different sources such as databases, web services or file systems. These data
+sources are subject to latency, and we'd prefer to query them efficiently.
+
+If we are just reading data, we can make a series of optimizations such as:
+
+ - batching requests to the same data source
+ - requesting independent data from different sources in parallel
+ - caching previously seen results
+
+However, if we mix these optimizations with the code that fetches the data
+we may end up trading clarity for performance. Furthermore, we are
+mixing low-level (optimization) and high-level (business logic with the data
+we read) concerns.
+
+Fetch is a library that allows your data fetches to be written in a concise,
+composable way while executing efficiently. You don't need to use any explicit
+concurrency construct but existing idioms: applicative for concurrency and
+monad for sequencing.
+
+# Installation
+
+First of all, add the following dependency to your SBT build file:
+
+```scala
+"org.fortysevendeg" %%% "fetch" % "0.1.0"
+```
+
+Now you'll have Fetch available both in Scala and Scala.js.
+
+# Usage
+
+In order to tell Fetch how to retrieve data, we must implement the `DataSource` typeclass.
+
+```scala
+trait DataSource[Identity, Result]{
+  def fetch(ids: List[Identity]): Eval[Map[Identity, Result]]
+}
+```
+
+It takes two type parameters:
+
+ - `Identity`: the identity we want to fetch (a `UserId` if we were fetching users)
+ - `Result`: the type of the data we retrieve (a `User` if we were fetching users)
+
+The `fetch` method takes a non empty list of identities and must return an `Eval` that will result
+in a map from identities to results. Accepting a list of identities gives Fetch the ability to batch requests to
+the same data source, and returning a mapping from identities to results Fetch can detect whenever an identity
+couldn't be fetched or no longer exists.
+
+Returning `Eval` makes it possible to defer evaluation with a concurrency monad when running a fetch. For example
+when using `Future` as your target monad when running a fetch the data source fetches will possibly run in a thread pool,
+depending on the implicit execution context you use.
+
+## Writing your first data source
+
+Now that we know about the `DataSource` typeclass, let's write our first data source! We'll be using Scala's
+`Future` as our concurrency monad for the examples. We'll write a little function for simulating latency and
```

Comment:
This is going to be confusing for users since they won't now this is not part of the actual runtime but a test function. IMO it does not belong in the docs specially not when they have not even seen the lib in action.